Lib/typing.py

```diff
@@ -1290,13 +1291,30 @@ def __call__(self, *args, **kwargs):
         if not self._inst:
             raise TypeError(f"Type {self._name} cannot be instantiated; "
                             f"use {self.__origin__.__name__}() instead")
-        result = self.__origin__(*args, **kwargs)
+        clas = self.__origin__
+        is_python_type = inspect.isfunction(clas.__new__)
```

Comment:
This is also wrong (continuing to use my empty `A` class):

```
>>> inspect.isfunction(A.__new__)
False
```

I would recommend to write tests to check that this does what you want it to do before you tweak the code more.